crypto/fipsmodule/bn/internal.h

```diff
@@ -185,8 +185,13 @@ void bn_mul_mont(BN_ULONG *rp, const BN_ULONG *ap, const BN_ULONG *bp,
 
 static inline void bn_umult_lohi(BN_ULONG *low_out, BN_ULONG *high_out,
                                  BN_ULONG a, BN_ULONG b) {
-#if defined(OPENSSL_X86_64) && defined(_MSC_VER) && !defined(__clang__)
+#if defined(_MSC_VER) && !defined(__clang__)
+#if defined(OPENSSL_X86_64)
   *low_out = _umul128(a, b, high_out);
+#elif defined(OPENSSL_AARCH64)
+  *low_out = a * b;
+  *high_out = __umulh(a, b);
+#endif
```

Comment:
Doesn't the `#else` branch immediately below already do the right thing? I don't see the advantage of this change.

Reply:
TBH I copied it from here without looking 😬 https://github.com/microsoft/winrtc/blob/ef004bdb791282e11cfbe3312cda31f30f5afbeb/patches_for_WebRTC_org/m84/4001-Arm64-is-a-thing-and-has-intrinsic-to-mul-two-64bit-.patch#L24

The only difference seems to be that this branch uses an [`__umulh`](https://docs.microsoft.com/en-us/cpp/intrinsics/umulh?view=msvc-160) intrinsic. I think we can remove it indeed 😅

Reply:
It seems like we need this because MSVC doesn't have `BN_ULLONG` for Aarch64 since that would be a 128-bit type. So no change needed here.